client/src/charts/legends/SelectAllControl.js

```diff
@@ -0,0 +1,22 @@
+import { Fragment } from 'react';
+import { TrivialTM } from '../../components/index.js';
+
+export const SelectAllControl = ({SelectAllOnClick, SelectNoneOnClick}) =>
+  <Fragment>
+    <TrivialTM k="select" />:
+    <span
+      onClick={SelectAllOnClick}
+      className="link-styled"
+      style={{margin: "0px 5px 0px 5px"}}
+    >
+      <TrivialTM k="all"/>
+    </span>
+|
+    <span
+      onClick={SelectNoneOnClick}
+      className="link-styled" 
+      style={{marginLeft: "5px"}}
+    >
+      <TrivialTM k="none"/>
+    </span> 
```

Comment:
So, these spans-as-buttons are going to need some adjustments for a11y purposes.

Screen readers aside, the spans currently can't be used by keyboard navigators on the standard site. Spans aren't tab-focusable by default, and even if they were you wouldn't be able to properly fire the onClick from the keyboard.

If you want to keep them as spans you need to:
  - give them `tabIndex={0}`
    - explanation: tab index sets the local tabbing order if non-zero, but we rarely use that (order in the DOM's the better way to control tab order). When zero, it makes any element tab focusable and puts adds i to the default tabbing order. Negative tab index can be used to remove tab navigation from an element that has it by default
  - add `onKeyDown={ ({keyCode}) => (keyCode === 13 || keyCode === 32) && SelectAllOnClick()}`, etc.
    - keyboard navigation expects both enter (13) space bar (32) to work on a button

Alternatively (preferable) is to use buttons instead of spans here. Buttons are tab-navigable by default AND they automatically trigger their onClick when a space or enterkey is pressed on them. You could use the `button-unstyled` class and a bit of css tweaking to keep them looking like the spans currently do.

Not as important here because I think these controls won't be anywhere on the a11y build, but the span's would also need `role='button'` for proper screen reader semantics.

I think the ultimately correct approach would be something like the following, although I don't usually go all out with groups and labels (probably should be)
```
<div role="group" aria-labelledby={unique_id}>
  <div id={unique_id}>
     <TrivialTM k="select" />:
  </div>
  <button onClick={SelectAllOnClick}>
     <TrivialTM k="all" />
   </span>
|
  <button onClick={SelectNoneOnClick}>
     <TrivialTM k="none" />
  </button> 
</div>
```
With whatever styling you want on that

Reply:
Ahh A11y😩

Reply:
Thanks for the explanations!